pkg/dependencies/helmfactory.go

```diff
@@ -0,0 +1,138 @@
+package dependencies
+
+import (
+	"context"
+	"sync"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	configcli "github.com/aws/eks-anywhere/pkg/config"
+	"github.com/aws/eks-anywhere/pkg/executables"
+	"github.com/aws/eks-anywhere/pkg/registrymirror"
+)
+
+// ExecutableBuilder builds the Helm executeble and returns it.
+type ExecutableBuilder interface {
+	BuildHelmExecutable(...executables.HelmOpt) *executables.Helm
+}
+
+// HelmFactoryOpt configures a HelmFactory.
+type HelmFactoryOpt func(*HelmFactory)
+
+// HelmFactory is responsible for creating and owning instances of Helm client.
+type HelmFactory struct {
+	client         cluster.Client
+	helmClient     *cluster.HelmClient
+	mu             sync.Mutex
+	builder        ExecutableBuilder
+	registryMirror *registrymirror.RegistryMirror
```

Comment:
are you using this?

Reply:
Yes, this option is set when building in the dependencies factory, mainly for the CLI flow and the reasons give [here](https://github.com/aws/eks-anywhere/pull/7170/files/6a9ada464d5672a4de7598e026c7ca40ce97a83d#r1425706153)

Reply:
yeah, let's finish the discussion there and then come back if necessary

Reply:
This was removed after addressing the issues mentioned in the linked thread.